README.md

```diff
@@ -13,8 +13,9 @@ that they are not required to be valid UTF-8.
 
 https://docs.rs/bstr
 
-The documentation includes a section on why you might want to use this crate.
+### Why use BString over String?
```

Comment:
Could you change this to "When should I use byte string?"

Reply:
I guess dropping the question mark would be more in line with existing READMEs and the intent should still be comparably clear.